### DISCUSSION TITLE
Supporting local mbtiles

### DISCUSSION THREAD

**Discussion by lionel008** [maplibre/maplibre-native#971] [Q&A]
Hello ,

     Currently i am using following code to add map in macos in swift
        let styleURL = URL(string: "https://api.maptiler.com/maps/streets-v2/style.json?key=\(mapTilerKey)")
        mapView.styleURL = styleURL
       Is there any example to load mbtiles file directly from https://data.maptiler.com/downloads/planet/?

**Comment by klokan** [comment]
MapLibre GL Native can load data from locally stored mbtiles files via "mbtiles://" protocol on source in the style definition.

For code snippet, see: https://github.com/maplibre/maplibre-gl-native/issues/17#issuecomment-883869688

Please contact MapTiler sales team for permission if you are keen to "Bundle or redistribute" MapTiler Data or map styles with your application. Usage from Cloud is of course possible directly via standard plans. We are happy to assist.

**Reply by klokan** [reply]
For traditional offline usage - where the app downloads region from server - check https://maplibre.org/maplibre-gl-native/ios/api/Offline%20Maps.html instead https://docs.maptiler.com/maplibre-gl-native-android/com.mapbox.mapboxsdk.offline/
